libshvbroker/src/rpc/masterbrokerconnection.cpp

```diff
@@ -170,7 +170,11 @@ void MasterBrokerConnection::onRpcDataReceived(shv::chainpack::Rpc::ProtocolType
 			else if (rq_id == m_masterBrokerIdRqId) {
 				m_masterBrokerIdRqId = 0;
 				cp::RpcValue rpc_val = decodeData(protocol_type, msg_data, 0);
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wuseless-cast"
```

Comment:
why the useless-cast is generated here?

Reply:
I think that it's because when referring to T&& variable by name, it becomes a T&. And casting T& to T is a useless cast. Also, having all these && arguments and std::move is a symptom of poor design. Half of these are probably are not even used and are possibly even hurtful to performance. Maybe I'll look into it tomorrow.

Reply:
I couldn't do anything about it really, other than making the copy as a local variable, which isn't any better IMHO.

Reply:
This is related https://gcc.gnu.org/bugzilla/show_bug.cgi?id=85043